portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -249,6 +285,48 @@ private void deleteComment(String id) {
     }
   }
 
+  /** Returns a blob key that identifies the uploaded file, or null if the user didn't upload a file. */
```

Comment:
Instead of returning `null`, which can sometimes cause `NullPointerExceptions`, we should return Optional.absent() when there is no value present.

You can use Optional<BlobKey> as the return type of this method.